googleprediction/gp-connector/gp-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/googleprediction-config/init.xml

```diff
@@ -16,24 +16,61 @@
-	<parameter name="project" description="The project available in GooglePrediction." />
 
-	<sequence>
-		<property name="uri.var.apiUrl" expression="$func:apiUrl" />
-		<property name="uri.var.accessToken" expression="$func:accessToken" />
-		<property name="Authorization" expression="fn:concat('Bearer ', get-property('uri.var.accessToken'))" scope="transport"
-			type="STRING" />
-		<property name="Content-Type" value="application/json" scope="transport" type="STRING" />
-		<property name="uri.var.project" expression="$func:project" />
+<template name="init" xmlns="http://ws.apache.org/ns/synapse"> <parameter name="refreshToken" description="Refresh Token"/>
+    <parameter name="clientId"
+               description="clientId of your App, given by YouTube when you registered your application"/>
+    <parameter name="clientSecret"
+               description="clientSecret of your App, given by prediction when you registered your application"/>
+    <parameter name="apiUrl" description="API Url to access the endpoint."/>
+    <parameter name="accessToken" description="API Url to access the endpoint."/>
+    <parameter name="project" description="API Url to access the endpoint."/>
+    <parameter name="refreshToken" description="refreshToken to access the endpoint."/>
+    <parameter name="accessTokenRegistryPath" description="API Url to access the endpoint."/>
+    <sequence>
+        <property name="uri.var.refreshToken" expression="$func:refreshToken"/>
+        <property name="uri.var.clientId" expression="$func:clientId"/>
+        <property name="uri.var.clientSecret" expression="$func:clientSecret"/>
+        <property name="uri.var.apiUrl" expression="$func:apiUrl"/>
+        <property name="uri.var.project" expression="$func:project"/>
 
-		<!-- If the apiUrl is not specified, set the default value as https://www.googleapis.com -->
-		<filter source="boolean(get-property('uri.var.apiUrl'))" regex="false">
-			<then>
-				<property name="uri.var.apiUrl" value="https://www.googleapis.com" />
-			</then>
-		</filter>
-	</sequence>
+        <property name="uri.var.accessToken" expression="$func:accessToken"/>
+        <property name="uri.var.accessTokenRegistryPath" expression="$func:accessTokenRegistryPath"/>
+        <header name="Transfer-Encoding" action="remove" scope="transport"/>
+        <filter source="boolean($ctx:uri.var.accessTokenRegistryPath)" regex="false">
+            <then>
+                <property name="uri.var.accessTokenRegistryPath"
+                          value="connectors/GooglePrediction/accessToken"/>
+
+            </then>
+            <else>
+                <property name="uri.var.accessTokenRegistryPath"
+                          expression="$ctx:uri.var.accessTokenRegistryPath"/>
+            </else>
+        </filter>
+        <filter source="boolean($ctx:uri.var.accessToken)" regex="false">
```

Comment:
Logic is wrong here. User will pass access token. We have to check in the second message flow read the token from registry.